src/main/cpp/include/query_operations/variant_operations.h

```diff
@@ -327,8 +327,13 @@ class VariantOperations {
       end_idx = r;
     }
     auto numerator = 1ull;
-    for (auto i=begin_idx; i<=static_cast<uint64_t>(n); ++i)
+    for (auto i=begin_idx; i<=static_cast<uint64_t>(n); ++i) {
+      auto numerator_bits = 64 - __builtin_clzll(numerator);
+      auto i_bits = 64 - __builtin_clzll(i);
+      if(numerator_bits + i_bits > 64) //this might sometimes falsely mark some products are overflowing
+	return UINT64_MAX;
       numerator *= i;
+    }
```

Comment:
Yeah, if it makes thing easier I'll just replace this logic with the mul/div check. I was worried that the division would reduce performance. But, in the larger scheme of things, it's probably minor.
This would also get rid of the gcc builtin portability issue.

Reply:
Yes, division is not efficient. As long as this is portable, I am fine.

Reply:
Unit test for nCr?

Reply:
Yes for nCr especially when it returns UINT64_MAX. If it is covered elsewhere(run.py), it is not necessary.